File: leetcode/easy/reverse-prefix-of-word.ts
// https://leetcode.com/problems/reverse-prefix-of-word/

// MINE
/*
function reversePrefix(word: string, ch: string): string {
    const i = word.indexOf(ch)
    return word.slice(0, i+1).split('').reverse().join('') + word.slice(i+1)
};
*/

// ANSWER: without reverse
function reversePrefix(word: string, ch: string): string {
  let output = '',
    idx = word.indexOf(ch)
  for (var i = idx; i >= 0; i--) {
    output = output + word.slice(i, i + 1)
  }

  output = output + word.slice(idx + 1)

  return output
}

File: leetcode/easy/reverse-prefix-of-word.py
# https://leetcode.com/problems/reverse-prefix-of-word/

class Solution:
    def reversePrefix(self, word: str, ch: str) -> str:
        # MINE
        '''
        output = ''

        if not ch in word:
            return word

        idx = word.index(ch)

        for i in range(idx, -1, -1):
            output += word[i]

        output += word[idx+1:]

        return output
        '''

        # ANSWER
        idx = word.find(ch)
        if idx:
            return word[:idx+1][::-1] + word[idx+1:]
        return word
